### PR TITLE
Feature: Show entity dependencies in debugger

### PR DESCRIPTION
**Pull request by akash-codemonk**
## Description
Show entity dependencies in the debugger

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Cypress test

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: feat/debugger-entity-relationships 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 51.56 **(-0.01)** | 32.64 **(-0.1)** | 29.57 **(-0.08)** | 52.04 **(-0.03)**
 :green_circle: | app/client/src/components/editorComponents/ApiResponseView.tsx | 50 **(0.68)** | 32.79 **(0)** | 0 **(0)** | 52.86 **(0.69)**
 :green_circle: | app/client/src/components/editorComponents/Debugger/DebuggerLogs.tsx | 29.79 **(1.53)** | 25 **(0)** | 0 **(0)** | 36.84 **(1.7)**
 :green_circle: | app/client/src/components/editorComponents/Debugger/DebuggerTabs.tsx | 55.17 **(3.32)** | 100 **(0)** | 0 **(0)** | 61.54 **(3.21)**
 :sparkles: :new: | **app/client/src/components/editorComponents/Debugger/EntityDependecies.tsx** | **31.37** | **40** | **0** | **34.04**
 :green_circle: | app/client/src/components/editorComponents/Debugger/helpers.tsx | 78.05 **(48.88)** | 78.57 **(69.48)** | 27.27 **(27.27)** | 76.47 **(45.7)**
 :sparkles: :new: | **app/client/src/components/editorComponents/Debugger/hooks.ts** | **26.19** | **0** | **0** | **24.66**
 :green_circle: | app/client/src/constants/messages.ts | 71.8 **(0.4)** | 100 **(0)** | 14.71 **(1.21)** | 75.52 **(0.34)**
 :green_circle: | app/client/src/pages/Editor/QueryEditor/EditorJSONtoForm.tsx | 50 **(0.78)** | 48.48 **(0)** | 0 **(0)** | 51.59 **(0.78)**</details>

### CODE REVIEW

File: app/client/src/components/editorComponents/Debugger/helpers.tsx [hunk 1]
```diff
@@ -97,3 +116,152 @@ export const usePagination = (data: Message[], itemsPerPage = 50) => {
 
   return { next, paginatedData };
 };
+
+export function getDependencies(
```
Comment (rishabhsaxena) [inline app/client/src/components/editorComponents/Debugger/helpers.tsx:120]: Should we rename it to `getDependenciesFromInverseDependencies`? As you see fit

File: app/client/src/components/editorComponents/Debugger/helpers.tsx [hunk 2]
```diff
@@ -97,3 +116,152 @@ export const usePagination = (data: Message[], itemsPerPage = 50) => {
...
+    });
+  });
+
+  return {
+    inverseDependencies: Array.from(inverseDependencies),
+    directDependencies: Array.from(directDependencies),
+  };
+}
+
+export const useSelectedEntity = () => {
```
Comment (rishabhsaxena) [inline app/client/src/components/editorComponents/Debugger/helpers.tsx:158]: We could create a separate `hooks` file within debugger for all hooks

**Review by rishabhsaxena — APPROVED** [review]
Few thoughts:
- Does the `no dependencies exist state` ui, after selecting an entity looks a bit empty?
- Do we have an easy way to show the values being referenced and for which field? So for example if Text1.text is `{{ Text2.text + Text3.text }}`, we could show the value of Text1.text, and under the list show items.
- Also should we mention triggers here as well? `onClick` / `onSelect`

@Debsourabh, adding you as a reviewer, you could checkout the deploy preview here: https://deploy-preview-4356--appsmith-editor-ce.netlify.app/

**Review by Debsourabh — COMMENTED** [review]
When a user goes from inspect entity to the said entity and comes back to the debugger they are in an error state by default. They should be back to the debugger with the Inspect Entity opened. 

Also when the user is going from canvas - debugger to API/Query debugger the same tab should be opened. Currently we default it to the error tab which is not ideal. 

image
Response from a Postgres DB is not displayed correctly.

**Comment by akash-codemonk** [comment]
> * Do we have an easy way to show the values being referenced and for which field? So for example if Text1.text is `{{ Text2.text + Text3.text }}`, we could show the value of Text1.text, and under the list show items.
> * Also should we mention triggers here as well? `onClick` / `onSelect`
Entity properties were being shown but it was decided to just show the entity names. Also for bindings with 2 or more lines it would probably not look great.

**Comment by akash-codemonk** [comment]
> When a user goes from inspect entity to the said entity and comes back to the debugger they are in an error state by default. They should be back to the debugger with the Inspect Entity opened.
> 
> Also when the user is going from canvas - debugger to API/Query debugger the same tab should be opened. Currently we default it to the error tab which is not ideal.

Will create a separate issue for this.

**Comment by akash-codemonk** [comment]
> * Does the `no dependencies exist state` ui, after selecting an entity looks a bit empty?

Since the new design is not yet there can we move ahead for now and update later when the designs are there?

**Review by rishabhsaxena — DISMISSED** [review]

**Review by rishabhsaxena — APPROVED** [review]

**Pull request merged**